signer/src/bitcoin/validation.rs

```diff
@@ -0,0 +1,434 @@
+    /// we do not do a check of whether we will accept it otherwise.
+    ///
+    /// This will only be `None` if we do not have a record of the deposit
+    /// request.
+    pub can_sign: Option<bool>,
+    /// Whether this signer accepted the deposit request or not. This
+    /// should only be `None` if we do not have a record of the deposit
+    /// request or if we cannot sign for the deposited funds.
+    pub is_accepted: Option<bool>,
+    /// The deposit amount
+    pub amount: u64,
+    /// The lock_time in the reclaim script
+    pub lock_time: LockTime,
+}
+
+impl DepositRequestReport {
+    /// Validate that the deposit request is okay given the report.
+    pub fn validate(self, chain_tip_height: u64) -> Result<(), BitcoinDepositInputError> {
+        let confirmed_block_height = match self.status {
+            // Deposit requests are only written to the database after they
+            // have been confirmed, so this means that we have a record of
+            // the request, but it has not been confirmed on the canonical
+            // bitcoin blockchain.
+            DepositRequestStatus::Unconfirmed => {
+                return Err(BitcoinDepositInputError::TxNotOnBestChain(self.outpoint));
+            }
+            // This means that we have a record of the deposit UTXO being
+            // spent in a sweep transaction that has been confirmed on the
+            // canonical bitcoin blockchain.
+            DepositRequestStatus::Spent => {
+                return Err(BitcoinDepositInputError::DepositUtxoSpent(self.outpoint));
+            }
+            // The deposit has been confirmed on the canonical bitcoin
+            // blockchain and remains unspent by us.
+            DepositRequestStatus::Confirmed(block_height) => block_height,
+        };
+
+        match self.can_sign {
+            // If we are here, we know that we have a record for the
+            // deposit request, but we have not voted on it yet, so we do
+            // not know if we can sign for it.
+            None => return Err(BitcoinDepositInputError::NoVote(self.outpoint)),
+            // In this case we know that we cannot sign for the deposit
+            // because it is locked with a public key where the current
+            // signer is not part of the signing set.
+            Some(false) => return Err(BitcoinDepositInputError::CannotSignUtxo(self.outpoint)),
+            // Yay.
+            Some(true) => (),
+        }
+        // If we are here then can_sign is Some(true) so is_accepted is
+        // Some(_). Let's check whether we rejected this deposit.
+        if self.is_accepted != Some(true) {
+            return Err(BitcoinDepositInputError::RejectedRequest(self.outpoint));
+        }
+
+        // We only sweep a deposit if the depositor cannot reclaim the
+        // deposit within the next DEPOSIT_LOCKTIME_BLOCK_BUFFER blocks.
+        let deposit_age = chain_tip_height.saturating_sub(confirmed_block_height);
```

Comment:
Does `DEPOSIT_LOCKTIME_BLOCK_BUFFER` represent something different from the const defined in [743](https://github.com/stacks-network/sbtc/pull/743/files#diff-e5ee0ea573d6265d825f3e3d4ca66fea42acc7fba5ff75ff04d4e49171aed216R29)?

Reply:
Nope, they represent the same thing.